Amazon.IonObjectMapper.Test/IonObjectSerializerTest.cs

```diff
@@ -619,5 +619,77 @@ private void AssertIsTruck(object actual)
         {
             Assert.AreEqual(actual.ToString(), TestObjects.nativeTruck.ToString());
         }
+
+        [TestMethod]
+        public void TestSerializingAndDeserializingPropertiesWithAccessModifiers()
+        {
+            var classWithProperties = new ClassWithProperties(PublicProperty: "PublicProperty", DefaultProperty: "DefaultProperty",
+                ProtectedProperty: "ProtectedProperty", ProtectedInternalProperty: "ProtectedInternalProperty", InternalProperty: "InternalProperty",
+                PrivateProperty: "PrivateProperty", PrivateProtectedProperty: "PrivateProtectedProperty");
+
```

Comment:
Add a check to test that serialized data has only public/itnernal/protected internal properties/

Reply:
Also add comment

Reply:
Added tests for serialization